src/test/rpc/Book_test.cpp

```diff
@@ -800,6 +800,460 @@ class Book_test : public beast::unit_test::suite
+            jvParams[jss::taker_pays][jss::currency] = "XRP";
+            jvParams[jss::taker_gets][jss::currency] = "USD";
+            jvParams[jss::taker_gets][jss::issuer]   = gw.human();
+            auto const jrr = env.rpc(
+                "json", "book_offers", to_string(jvParams)) [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "invalidParams");
+            BEAST_EXPECT(jrr[jss::error_message] ==
+                "Invalid field 'limit', not unsigned integer.");
+        }
+
+        {
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = "validated";
+            jvParams[jss::taker_pays][jss::currency] = "USD";
+            jvParams[jss::taker_pays][jss::issuer]   = gw.human();
+            jvParams[jss::taker_gets][jss::currency] = "USD";
+            auto const jrr = env.rpc(
+                "json", "book_offers", to_string(jvParams)) [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "dstIsrMalformed");
+            BEAST_EXPECT(jrr[jss::error_message] ==
+                "Invalid field 'taker_gets.issuer', "
+                "expected non-XRP issuer.");
+        }
+
+        {
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = "validated";
+            jvParams[jss::taker_pays][jss::currency] = "USD";
+            jvParams[jss::taker_pays][jss::issuer]   = gw.human();
+            jvParams[jss::taker_gets][jss::currency] = "XRP";
+            jvParams[jss::taker_gets][jss::issuer]   = gw.human();
+            auto const jrr = env.rpc(
+                "json", "book_offers", to_string(jvParams)) [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "dstIsrMalformed");
+            BEAST_EXPECT(jrr[jss::error_message] ==
+                "Unneeded field 'taker_gets.issuer' "
+                "for XRP currency specification.");
+        }
+
+    }
+
+    void testBookOfferLimits(void)
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account gw {"gw"};
+        env.fund(XRP(20000), gw);
+        env.close();
+        auto USD = gw["USD"];
+
+        env(offer(gw, XRP(500), USD(100)));
+        env(offer(gw, XRP(100), USD(2)));
+        env(offer(gw, XRP(500), USD(101)));
+        env(offer(gw, XRP(500), USD(99)));
+        env(offer(gw, XRP(50), USD(10)));
+        env(offer(gw, XRP(50), USD(9)));
+        env.close();
+
+        Json::Value jvParams;
+        jvParams[jss::limit] = 1;
+        jvParams[jss::ledger_index] = "validated";
+        jvParams[jss::taker_pays][jss::currency] = "XRP";
+        jvParams[jss::taker_gets][jss::currency] = "USD";
+        jvParams[jss::taker_gets][jss::issuer] = gw.human();
+        auto const jrr =
+            env.rpc("json", "book_offers", to_string(jvParams)) [jss::result];
+        BEAST_EXPECT(jrr[jss::offers].isArray());
+        BEAST_EXPECT(jrr[jss::offers].size() == 1);
+        // NOTE - a marker field is not returned for this method
+    }
```

Comment:
I pushed branch with a fix that you can cherry-pick. If you decide not to address the issue in this PR, please let me know and I'll submit a PR for the bug. https://github.com/miguelportilla/rippled/commits/book_limit